packages/create/src/cli/change-data-source/change-data-source.ts

```diff
@@ -0,0 +1,106 @@
+import type { Command } from 'commander';
+import type { AdpProjectData, DataSourceData, PromptDefaults } from '@sap-ux/adp-tooling';
+import {
+    generateChange,
+    ChangeType,
+    ChangeDataSourcePrompts,
+    getTargetDataSources,
+    getTargetODataAnnotations,
+    getDataSourcesDictionary,
+    getDataServicesWithURI
+} from '@sap-ux/adp-tooling';
+import { prompt } from 'inquirer';
+import { createForAbap } from '../../../../axios-extension';
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { UI5Config } from '@sap-ux/ui5-config';
+
+/**
+ * Add a new sub-command to change the data source of an adaptation project to the given command.
+ *
+ * @param {Command} cmd - The command to add the change-data-source sub-command to.
+ */
+export function addChangeDataSourceCommand(cmd: Command): void {
```

Comment:
Check the other commands, we usually also have a mode that will not write the change but just shows the diffs so people can see what would be happening before they do it.

Reply:
Fixed with commit: f52e8525f3561ddc294690f963bd287bdb42d811